### PR TITLE
Make our link check action open a new issue if none is open

### PR DESCRIPTION
**Pull request by choldgraf**
This updates the our broken link workflow to open new issues none are already open, instead of re-opening an old issue, per @damianavila's feedback

### CODE REVIEW

File: .github/workflows/documentation-link-check.yaml [hunk 1]
```diff
@@ -85,14 +92,20 @@ jobs:
                 out.append(f"{iline}\n")
           warnings.write_text("\n".join(out))
 
+      # Return a list of open issues with the title we use for broken links
+      # If one exists, we update that issue. If not, then we create a new issue.
+      - name: Check for an open issue
+        run: |
+          ISSUE_NUMBER=$( gh issue list --search "${{ inputs.issue_title }}" | awk '{print $1}' )
```
Comment (damianavila) [inline .github/workflows/documentation-link-check.yaml:95]: `gh issue list` will list open issues by default, am I right?

Comment (choldgraf) [reply .github/workflows/documentation-link-check.yaml:95]: yep

File: .github/workflows/documentation-link-check.yaml [hunk 2]
```diff
@@ -85,14 +92,20 @@ jobs:
...
+          echo "ISSUE_NUMBER=$ISSUE_NUMBER" >> $GITHUB_ENV
+
+      # Use the warnings.txt content to populate a new issue.
+      # If ISSUE_NUMBER is None, a new issue will be created.
+      # otherwise a pre-existing issue will be updated.
```
Comment (damianavila) [inline .github/workflows/documentation-link-check.yaml:100]: How `ISSUE_NUMBER` env variable goes into `inputs.issue_number`? I think I am missing some context (or more details about GH actions).

Comment (choldgraf) [reply .github/workflows/documentation-link-check.yaml:100]: oops good catch - it should actually be `env.issue_number` and it is defined in the step above

File: .github/workflows/documentation-link-check.yaml [hunk 3]
```diff
@@ -85,14 +86,20 @@ jobs:
...
       - name: Create issue from errors output
         if: steps.buildDocs.outcome == 'failure'
         uses: peter-evans/create-issue-from-file@v4.0.0
         with:
-          issue-number: ${{ inputs.issue_number }}
-          title: Broken links in documentation
+          issue-number: ${{ env.issue_number }}
```
Comment (choldgraf) [inline .github/workflows/documentation-link-check.yaml]:
```suggestion
          issue-number: ${{ env.ISSUE_NUMBER }}
```

**Comment by choldgraf** [comment]
Ok I'm gonna try merging this so we can test it out

**Pull request merged**